homeassistant/components/zeroconf/__init__.py

```diff
@@ -45,3 +81,27 @@
     hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_zeroconf)
 
     return True
+
+
+def info_from_service(service):
+    """Return most important info from mDNS entries."""
+    properties = {}
+
+    for key, value in service.properties.items():
+        if isinstance(value, bytes):
+            value = value.decode('utf-8')
+        properties[key.decode('utf-8')] = value
+
+    address = service.address or service.address6
+
+    info = {
```

Comment:
We should also have access to the zeroconf address here, right? It would be preferable that integrations try to connect to that address, as it will always point at latest IP.

Reply:
I can't see any Zeroconf address. Do you mean the e.g. tv.lg_webos.local. address?

Reply:
I think balloob means the full hostname (e.g., `hassio.local`), not the service name, but that's already in `ATTR_HOSTNAME`, although I'm not sure if those will resolve per default on any other OS besides OSX?

However, I agree with the notion that as much information as possible should be extracted; this includes the type (maybe an integration supports multiple different implementations with different handling? without knowing which type it was, it may get hacky to see what it should do). You can see the available info here: https://github.com/frawau/aiozeroconf/blob/master/aiozeroconf/aiozeroconf.py#L1253

Reply:
I guess saving type and name can't be bad if a component subscribes to multiple service types?

{'type': '_axis-video._tcp.local.', 'name': 'AXIS M1065-L - ACCC8E9C8ABC._axis-video._tcp.local.', 'address': b'\n\x00\x00,', 'address6': None, 'port': 80, 'weight': 0, 'priority': 0, 'server': 'axis-accc8e9c8abc.local.', '_properties': {b'macaddress': b'ACCC8E9C8ABC'}, 'text': b'\x17macaddress=ACCC8E9C8ABC'}

Reply:
the more data the better. Just know that we can't ever change the data type again without a breaking change, which is pretty much a no-go.

Reply:
Added type and name